erts/emulator/beam/erl_bif_binary.c

```diff
@@ -319,16 +319,104 @@ static void dump_ac_node(ACNode *node, int indent, int ch);
  * The needed size of binary data for a search structure - given the
  * accumulated string lengths.
  */
-#define BM_SIZE(StrLen) 	      /* StrLen: length of searchstring */ \
-((MYALIGN(sizeof(Sint) * (StrLen))) + /* goodshift array */                \
- MYALIGN(StrLen) +                    /* searchstring saved */             \
- (MYALIGN(sizeof(BMData))))           /* Structure */
+#define BM_SIZE_SINGLE()    /* Single byte search string */ \
+(MYALIGN(1) +               /* searchstring saved */        \
```

Comment:
Should this be sizeof(byte)?

Reply:
In the snippet above, this is equivalent to `MYALIGN(StrLen)`. In our case `StrLen` is 1. So I thought about passing 1. So maybe above should be `MYALIGN(StrLen * byte())` too?